google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/scaladsl/GooglePubSub.scala

```diff
@@ -67,6 +68,7 @@ protected[pubsub] trait GooglePubSub {
   def subscribeFlow(subscription: String, config: PubSubConfig): Flow[Done, ReceivedMessage, Future[NotUsed]] = {
     flow(config)(httpApi.pull(subscription, config.pullReturnImmediately, config.pullMaxMessagesPerInternalBatch))
       .mapConcat(_.receivedMessages.getOrElse(Seq.empty[ReceivedMessage]).toIndexedSeq)
+      .buffer(1, OverflowStrategy.backpressure) // Needed to prevent a deadlock in IntegrationSpec
```

Comment:
Anyone know a better fix for this deadlock?

Reply:
I'm not sure what failure you were experiencing in `IntegrationSpec`, but when I added a line to cancel the test probe the test passed successfully (without the buffer). The other test doesn't have an issue because `Sink.head` cancels the stream when the first element arrives.

Reply:
The test `"receive a published message and acknowledge it"` was failing the `assertAllStagesStopped`. Which line did you change?

Reply:
I cancelled the probe with `result2.cancel()`.

https://github.com/armanbilge/alpakka/compare/wip-google-common...seglo:seglo/wip-google-common?expand=1

Reply:
Thanks. I was confused why this test was passing without the probe cancel before this PR.